x/wasm-storage/keeper/abci.go

```diff
@@ -32,3 +62,50 @@ func (k Keeper) ProcessExpiredWasms(ctx sdk.Context) error {
 	}
 	return nil
 }
+
+func (k Keeper) ExecuteTally(ctx sdk.Context) error {
+	// 1. Get contract address.
+	contractAddr, err := k.ProxyContractRegistry.Get(ctx)
+	if contractAddr == "" || errors.Is(err, collections.ErrNotFound) {
+		k.Logger(ctx).Debug("proxy contract address not registered")
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	// 2. Fetch tally-ready DRs.
+	// TODO: json marshal request struct?
+	queryRes, err := k.wasmViewKeeper.QuerySmart(ctx, sdk.MustAccAddressFromBech32(contractAddr), []byte(`{"get_data_requests_by_status":{"status": "tallying"}}`))
+	if err != nil {
+		return err
+	}
+	var tallyingList TallyingList
+	err = json.Unmarshal(queryRes, &tallyingList)
+	if err != nil {
+		return err
+	}
+
+	// 3. Loop through the list to execute tally.
+	// TODO: is it ok to use a map?
+	for id := range tallyingList {
+		// TODO: filtering
+
+		tallyWasm, err := k.DataRequestWasm.Get(ctx, tallyingList[id].TallyBinaryID)
+		if err != nil {
+			return fmt.Errorf("failed to get tally wasm for DR ID %s: %w", id, err)
+		}
+
+		result := tallyvm.ExecuteTallyVm(tallyWasm.Bytecode, []string{"1", "2"}, map[string]string{
+			"PATH": os.Getenv("SHELL"),
+		})
+		fmt.Println(result)
+	}
+
+	// 4. Post result.
+	// msg := []byte("{\"data_requests\": {}}")
```

Comment:
It's not relevant yet; just linking the appropriate rust for the json.

https://github.com/sedaprotocol/seda-common-rs/blob/main/src/msgs/data_requests/sudo/post_result.rs#L6

https://github.com/sedaprotocol/seda-common-rs/blob/main/src/msgs/data_requests/types.rs#L112

Reply:
Well noted.